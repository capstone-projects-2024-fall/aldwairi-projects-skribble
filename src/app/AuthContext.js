import React, { createContext, useState } from "react";

/**
 * AuthContext is a context that provides authentication-related state
 * (such as the session token) and functions (like setting the session token)
 * to the components within the application.
 *
 * @type {React.Context<{ sessionToken: string | null, setSessionToken: React.Dispatch<React.SetStateAction<string | null>> }>}
 */
export const AuthContext = createContext();

/**
 * AuthProvider component is a wrapper that provides the authentication context
 * to the children components. It manages the session token state, allowing
 * components to access and update it.
 * 
 * - Provides the `sessionToken` and `setSessionToken` via context.
 * - Any component wrapped in this provider can access or modify the authentication state.
 * 
 * @param {{ children: React.ReactNode }} param0 - The child components to be wrapped.
 * @param {React.ReactNode} param0.children - The child elements that will have access to the authentication context.
 * @returns {JSX.Element} The rendered AuthProvider component with the context provider.
 */
export const AuthProvider = ({ children }) => {
  const [sessionToken, setSessionToken] = useState(null);

  return (
    <AuthContext.Provider value={{ sessionToken, setSessionToken }}>
      {children}
    </AuthContext.Provider>
  );
};
