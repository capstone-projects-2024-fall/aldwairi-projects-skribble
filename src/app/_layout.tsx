import React from "react";
import { Stack } from "expo-router";

export default function RootLayout() {
  return (
    <Stack>
      <Stack.Screen name="index" options={{ title: "LogIn" }} />
      <Stack.Screen name="home" options={{ title: "Home" }} />
    </Stack>
  );
}
