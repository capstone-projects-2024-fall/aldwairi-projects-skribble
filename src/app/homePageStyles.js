import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  homeContainer: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
  },
  logoContainer: {
      position: 'absolute',
      alignItems: 'center',
  },
  logoContainerWeb: {
      top: 40,
  },
  logoContainerMobile: {
      top: 60,
  },
  logo: {
      resizeMode: 'contain',
  },
  avatar: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  buttonContainer: {
      flexDirection: 'row',
      flexWrap: 'wrap',
      justifyContent: 'center',
      marginTop: 20,
  },
  buttonStyle: {
      padding: 18,
      margin: 18,
      borderRadius: 18,
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
      borderWidth: 3,
  },
  buttonText: {
      color: 'black',
      fontSize: 20,
      marginRight: 14,
      fontWeight: 'bold',
  },
  icon: {
      width: 32,
      height: 32,
      marginLeft: 10,
  },
});

export default styles;