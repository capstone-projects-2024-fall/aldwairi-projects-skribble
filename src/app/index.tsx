import React, { useState } from "react";
import { Text, View, TextInput, TouchableOpacity, Alert } from "react-native";
import neo4j from "neo4j-driver";
import { useRouter } from "expo-router";
import styles from "./indexStyles";
import createNeo4jDriver from './utils/databaseSetUp';

export default function LogIn() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [birthday, setBirthday] = useState("");
  const [parentEmail, setParentEmail] = useState("");
  const [isSignUp, setIsSignUp] = useState(true);
  const [error, setError] = useState(""); // State for error message
  const router = useRouter();

  // Set up the Neo4j driver
  const driver = createNeo4jDriver();

  const handleAuthToggle = () => {
    setIsSignUp(!isSignUp);
    setError(""); // Clear error when switching between Sign Up and Sign In
  };

  const handleAuth = async () => {
    const session = driver.session();
  
    if (isSignUp) {
      if (!email || !password || !birthday) {
        setError("Email, password, and birthday cannot be blank.");
        return;
      }

      const birthDate = new Date(birthday);
      const age = new Date().getFullYear() - birthDate.getFullYear();
      const needsParentalControls = age < 13;

      if (needsParentalControls && !parentEmail) {
        setError("Parent's email is required for users under 13.");
        return;
      }
  
      try {
        // Create a new user or ensure the user exists
        const result = await session.run(
          `MERGE (u:User {email: $email})
           ON CREATE SET 
             u.password = $password,
             u.name = $name,
             u.streak = 0,
             u.coins = 0,
             u.exp = 0,
             u.backgroundColor = '#FFFFFF',
             u.birthday = $birthday,
             u.parentEmail = $parentEmail,
             u.needsParentalControls = $needsParentalControls
           RETURN u`,
          { email, password, name: "New User", birthday, parentEmail, needsParentalControls }
        );
  
        if (result.summary.counters.containsUpdates()) {
          console.log("User signed up successfully.");
          router.push("/homePage");
        } else {
          setError("User already exists.");
        }
      } catch (error) {
        console.error("Failed to sign up user", error);
        setError("An error occurred while signing up. Please try again.");
      } finally {
        await session.close();
        await driver.close();
      }
    } else {
      try {
        // Validate existing user credentials
        const result = await session.run(
          `MATCH (u:User {email: $email, password: $password})
           RETURN u`,
          { email, password }
        );
  
        if (result.records.length > 0) {
          console.log("User signed in successfully.");
          router.push("/homePage");
        } else {
          setError("Invalid email or password.");
        }
      } catch (error) {
        console.error("Failed to sign in user", error);
        setError("An error occurred while signing in. Please try again.");
      } finally {
        await session.close();
        await driver.close();
      }
    }
  };

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{isSignUp ? "Sign Up" : "Sign In"}</Text>

      <TextInput
        style={styles.input}
        placeholder="Email"
        placeholderTextColor="#888888"
        keyboardType="email-address"
        value={email}
        onChangeText={(text) => {
          setEmail(text);
          setError(""); // Clear error message when user starts typing
        }}
      />
      <TextInput
        style={styles.input}
        placeholder="Password"
        placeholderTextColor="#888888"
        secureTextEntry
        value={password}
        onChangeText={(text) => {
          setPassword(text);
          setError(""); // Clear error message when user starts typing
        }}
      />
      {isSignUp && (
        <>
          <TextInput
            style={styles.input}
            placeholder="Birthday (YYYY-MM-DD)"
            placeholderTextColor="#888888"
            value={birthday}
            onChangeText={(text) => {
              setBirthday(text);
              setError(""); // Clear error message when user starts typing
            }}
          />
          <TextInput
            style={styles.input}
            placeholder="Parent's Email (if under 13)"
            placeholderTextColor="#888888"
            keyboardType="email-address"
            value={parentEmail}
            onChangeText={(text) => {
              setParentEmail(text);
              setError(""); // Clear error message when user starts typing
            }}
          />
        </>
      )}

      {/* Display error message */}
      {error ? <Text style={styles.errorText}>{error}</Text> : null}

      <TouchableOpacity onPress={handleAuth} style={styles.button}>
        <Text style={styles.buttonText}>{isSignUp ? "Sign Up" : "Sign In"}</Text>
      </TouchableOpacity>

      <TouchableOpacity onPress={handleAuthToggle}>
        <Text style={styles.toggleText}>
          {isSignUp ? "Already have an account? Sign In" : "New user? Sign Up"}
        </Text>
      </TouchableOpacity>
    </View>
  );
}