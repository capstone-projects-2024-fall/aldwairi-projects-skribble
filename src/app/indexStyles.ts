import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    padding: 20,
  },
  title: {
    fontSize: 24,
    marginBottom: 20,
  },
  input: {
    width: "100%",
    height: 50,
    borderColor: "#ddd",
    borderWidth: 1,
    paddingHorizontal: 10,
    borderRadius: 5,
    marginBottom: 15,
  },
  button: {
    backgroundColor: "#007bff",
    padding: 15,
    borderRadius: 5,
    width: "100%",
    alignItems: "center",
  },
  buttonText: {
    color: "#fff",
    fontSize: 16,
  },
  toggleText: {
    color: "#007bff",
    marginTop: 15,
    textAlign: "center",
  },
  errorText: {
    color: "red",
    fontSize: 14,
    marginTop: 10,
    textAlign: "center",
  },
});

export default styles;
