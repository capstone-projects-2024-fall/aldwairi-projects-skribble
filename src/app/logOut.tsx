import AsyncStorage from '@react-native-async-storage/async-storage';
import { router, useRouter } from 'expo-router';

export const logOut = async () => {
    try {
        // Remove the current user data from AsyncStorage
        await AsyncStorage.removeItem('currentUser');
        console.log('User logged out successfully.');

        // Navigate back to the sign-in page
        router.push('/');
    } catch (error) {
        console.error('Failed to log out:', error);
    }
};
