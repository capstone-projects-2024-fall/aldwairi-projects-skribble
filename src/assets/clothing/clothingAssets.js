// category images
import bottoms from './bottomsCategory.png'
import tops from './topsCategory.png'
import outfits from './outfitsCategory.png'
import shoes from './shoesCategory.png'
import accessories from './accessoriesCategory.png'

// bottoms
import blackPants from './blackPants.png'
import blueWavyPants from './blueWavyPants.png'
import checkerSkirt from './checkerSkirt.png'
import greenSkirt from './greenSkirt.png'
import jeans from './jeans.png'
import plaidSkirt from './plaidSkirt.png'
import purpleSkirt from './purpleSkirt.png'
import splatterPants from './splatterPants.png'

// tops
import blueJacket from './blueJacketWithShirt.png'
import blueOmbreShirt from './blueOmbreShirt.png'
import blueSkribbleShirt from './blueSkribbleShirt.png'
import blueTankShirt from './blueTankShirt.png'
import brownJacket from './brownJacketWithShirt.png'
import greenSkribbleShirt from './greenSkribbleShirt.png'
import tropicalShirt from './greenTropicalShirt.png'
import pinkSkribbleShirt from './pinkSkribbleShirt.png'
import pinkTankShirt from './pinkTankShirt.png'
import purpleShirt from './purpleShirt.png'
import whiteButtonUp from './whiteButtonUpShirt.png'
import plaidShirt from './plaidShirt.png'

// full outfits
import blueDress from './blueDress.png'
import greenDress from './greenDress.png'
import jeanOveralls from './jeanOveralls.png'
import pinkDress from './pinkDress.png'
import pinkOveralls from './pinkOveralls.png'
import purpleDress from './purpleDress.png'
import flowerDress from './purpleFlowerDress.png'
import sageDress from './sageDress.png'
import sunflowerDress from './sunflowerDress.png'
import sunflowerOveralls from './sunflowerOveralls.png'
import tulipDress from './tulipDress.png'
import yellowDress from './yellowDress.png'

// shoes
import blackBoots from './blackBoots.png'
import blackShoes from './blackShoes.png'
import blueSocks from './blueSocks.png'
import pinkSocks from './pinkSocks.png'
import purpleBoots from './purpleBoots.png'
import yellowShoes from './yellowShoes.png'

// accessories
import blueHat from './blueHat.png'
import blueScarf from './blueScarf.png'
import crown from './crown.png'
import greenBdayHat from './greenBirthdayHat.png'
import greenTie from './greenTie.png'
import necklace from './necklace.png'
import pinkBow from './pinkBow.png'
import pinkHat from './pinkHat.png'
import purpleBdayHat from './purpleBirthdayHat.png'
import redBow from './redBow.png'
import redScarf from './redScarf.png'
import watch from './watch.png'

// load 
export const assets = {
    bottoms,
    tops,
    outfits,
    shoes,
    accessories,
    blackPants,
    blueWavyPants,
    checkerSkirt,
    greenSkirt,
    jeans,
    plaidSkirt,
    purpleSkirt,
    splatterPants,
    blueJacket,
    blueOmbreShirt,
    blueSkribbleShirt,
    blueTankShirt,
    brownJacket,
    greenSkribbleShirt,
    tropicalShirt,
    pinkSkribbleShirt,
    pinkTankShirt,
    purpleShirt,
    whiteButtonUp,
    plaidShirt,
    blueDress,
    greenDress,
    jeanOveralls,
    pinkDress,
    pinkOveralls,
    purpleDress,
    flowerDress,
    sageDress,
    sunflowerDress,
    sunflowerOveralls,
    tulipDress,
    yellowDress,
    blackBoots,
    blackShoes,
    blueSocks,
    pinkSocks,
    purpleBoots,
    yellowShoes,
    blueHat,
    blueScarf,
    crown,
    greenBdayHat,
    greenTie,
    necklace,
    pinkBow,
    pinkHat,
    purpleBdayHat,
    redBow,
    redScarf,
    watch
}

// clohing category
export const category_list = [
    {
        category_id: "bottoms",
        category_image: bottoms
    },
    {
        category_id: "tops",
        category_image: tops
    },
    {
        category_id: "full outfits",
        category_image: outfits
    },
    {
        category_id: "shoes",
        category_image: shoes
    },
    {
        category_id: "accessories",
        category_image: accessories
    }
]

// clothing items list
export const clothes_list = [
    {
        _id: "1",
        category: "bottoms",
        name: "black pants",
        image: blackPants,
        price: 5.00,
        quantity: 0
    },
    {
        _id: "2",
        category: "bottoms",
        name: "jeans",
        image: jeans,
        price: 10.00,
        quantity: 0
    },
    {
        _id: "3",
        category: "bottoms",
        name: "purple skirt",
        image: purpleSkirt,
        price: 15.00,
        quantity: 0
    },
    {
        _id: "4",
        category: "bottoms",
        name: "green skirt",
        image: greenSkirt,
        price: 15.00,
        quantity: 0
    },
    {
        _id: "5",
        category: "bottoms",
        name: "blue wavy pants",
        image: blueWavyPants,
        price: 20.00,
        quantity: 0
    },
    {
        _id: "6",
        category: "bottoms",
        name: "plaid skirt",
        image: plaidSkirt,
        price: 25.00,
        quantity: 0
    },
    {
        _id: "7",
        category: "bottoms",
        name: "checker skirt",
        image: checkerSkirt,
        price: 30.00,
        quantity: 0
    },
    {
        _id: "8",
        category: "bottoms",
        name: "splatter paint pants",
        image: splatterPants,
        price: 35.00,
        quantity: 0
    },
    {
        _id: "9",
        category: "tops",
        name: "blue Skribble shirt",
        image: blueSkribbleShirt,
        price: 5.00,
        quantity: 0
    },
    {
        _id: "10",
        category: "tops",
        name: "green Skribble shirt",
        image: greenSkribbleShirt,
        price: 5.00,
        quantity: 0
    },
    {
        _id: "11",
        category: "tops",
        name: "pink Skribble shirt",
        image: pinkSkribbleShirt,
        price: 5.00,
        quantity: 0
    },
    {
        _id: "12",
        category: "tops",
        name: "pink tank top",
        image: pinkTankShirt,
        price: 10.00,
        quantity: 0
    },
    {
        _id: "13",
        category: "tops",
        name: "blue tank top",
        image: blueTankShirt,
        price: 10.00,
        quantity: 0
    },
    {
        _id: "14",
        category: "tops",
        name: "tropical shirt",
        image: tropicalShirt,
        price: 15.00,
        quantity: 0
    },
    {
        _id: "15",
        category: "tops",
        name: "purple cheveron shirt",
        image: purpleShirt,
        price: 20.00,
        quantity: 0
    },
    {
        _id: "16",
        category: "tops",
        name: "white button up shirt",
        image: whiteButtonUp,
        price: 25.00,
        quantity: 0
    },
    {
        _id: "17",
        category: "tops",
        name: "plaid shirt",
        image: plaidShirt,
        price: 30.00,
        quantity: 0
    },
    {
        _id: "18",
        category: "tops",
        name: "white shirt with blue jacket",
        image: blueJacket,
        price: 35.00,
        quantity: 0
    },
    {
        _id: "19",
        category: "tops",
        name: "white shirt with brown jacket",
        image: brownJacket,
        price: 35.00,
        quantity: 0
    },
    {
        _id: "20",
        category: "tops",
        name: "ombre shirt",
        image: blueOmbreShirt,
        price: 40.00,
        quantity: 0
    },
    {
        _id: "21",
        category: "full outfits",
        name: "pink dress",
        image: pinkDress,
        price: 10.00,
        quantity: 0
    },
    {
        _id: "22",
        category: "full outfits",
        name: "blue dress",
        image: blueDress,
        price: 10.00,
        quantity: 0
    },
    {
        _id: "23",
        category: "full outfits",
        name: "green dress",
        image: greenDress,
        price: 10.00,
        quantity: 0
    },
    {
        _id: "24",
        category: "full outfits",
        name: "yellow dress",
        image: yellowDress,
        price: 15.00,
        quantity: 0
    },
    {
        _id: "25",
        category: "full outfits",
        name: "sage dress",
        image: sageDress,
        price: 15.00,
        quantity: 0
    },
    {
        _id: "26",
        category: "full outfits",
        name: "purple dress",
        image: purpleDress,
        price: 15.00,
        quantity: 0
    },
    {
        _id: "27",
        category: "full outfits",
        name: "jean overalls",
        image: jeanOveralls,
        price: 20.00,
        quantity: 0
    },
    {
        _id: "28",
        category: "full outfits",
        name: "pink patterned overalls",
        image: pinkOveralls,
        price: 25.00,
        quantity: 0
    },
    {
        _id: "29",
        category: "full outfits",
        name: "sunflower overalls",
        image: sunflowerOveralls,
        price: 25.00,
        quantity: 0
    },
    {
        _id: "30",
        category: "full outfits",
        name: "flower dress",
        image: flowerDress,
        price: 30.00,
        quantity: 0
    },
    {
        _id: "31",
        category: "full outfits",
        name: "tulip dress",
        image: tulipDress,
        price: 35.00,
        quantity: 0
    },
    {
        _id: "32",
        category: "full outfits",
        name: "sunflower dress",
        image: sunflowerDress,
        price: 40.00,
        quantity: 0
    },
    {
        _id: "33",
        category: "shoes",
        name: "blue socks",
        image: blueSocks,
        price: 5.00,
        quantity: 0
    },
    {
        _id: "34",
        category: "shoes",
        name: "pink socks",
        image: pinkSocks,
        price: 5.00,
        quantity: 0
    },
    {
        _id: "35",
        category: "shoes",
        name: "black shoes",
        image: blackShoes,
        price: 10.00,
        quantity: 0
    },
    {
        _id: "36",
        category: "shoes",
        name: "yellow shoes",
        image: yellowShoes,
        price: 15.00,
        quantity: 0
    },
    {
        _id: "37",
        category: "shoes",
        name: "black boots",
        image: blackBoots,
        price: 20.00,
        quantity: 0
    },
    {
        _id: "38",
        category: "shoes",
        name: "purple shoes",
        image: purpleBoots,
        price: 25.00,
        quantity: 0
    },
    {
        _id: "39",
        category: "accessories",
        name: "pink bow",
        image: pinkBow,
        price: 5.00,
        quantity: 0
    },
    {
        _id: "40",
        category: "accessories",
        name: "red bow",
        image: redBow,
        price: 5.00,
        quantity: 0
    },
    {
        _id: "41",
        category: "accessories",
        name: "blue snow hat",
        image: blueHat,
        price: 10.00,
        quantity: 0
    },
    {
        _id: "42",
        category: "accessories",
        name: "pink snow hat",
        image: pinkHat,
        price: 10.00,
        quantity: 0
    },
    {
        _id: "43",
        category: "accessories",
        name: "red scarf",
        image: redScarf,
        price: 15.00,
        quantity: 0
    },
    {
        _id: "44",
        category: "accessories",
        name: "blue scarf",
        image: blueScarf,
        price: 15.00,
        quantity: 0
    },
    {
        _id: "45",
        category: "accessories",
        name: "green bday hat",
        image: greenBdayHat,
        price: 20.00,
        quantity: 0
    },
    {
        _id: "46",
        category: "accessories",
        name: "purple bday hat",
        image: purpleBdayHat,
        price: 20.00,
        quantity: 0
    },
    {
        _id: "47",
        category: "accessories",
        name: "green tie",
        image: greenTie,
        price: 25.00,
        quantity: 0
    },
    {
        _id: "48",
        category: "accessories",
        name: "necklace",
        image: necklace,
        price: 30.00,
        quantity: 0
    },
    {
        _id: "49",
        category: "accessories",
        name: "watch",
        image: watch,
        price: 35.00,
        quantity: 0
    },
    {
        _id: "50",
        category: "accessories",
        name: "crown",
        image: crown,
        price: 40.00,
        quantity: 0
    },
]