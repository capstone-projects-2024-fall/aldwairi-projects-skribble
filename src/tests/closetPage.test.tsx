import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import ClosetPage from '../app/closetPage/closetPage';
import { AuthContext } from '../app/AuthContext';
import { category_list, clothes_list } from '../assets/clothing/clothingAssets';

// mock createNeo4jDriver
jest.mock('../app/utils/databaseSetUp', () => ({
  __esModule: true,
  default: jest.fn().mockReturnValue({
    session: jest.fn().mockReturnValue({
      run: jest.fn().mockResolvedValue({
        records: [
          {
            get: jest.fn((key) => {
              switch (key) {
                case 'name':
                  return 'New User';
                case 'avatarImage':
                  return '1';
                case 'ownedItems':
                  return ['1', '2']; // Mocked owned items
                case 'wornItems':
                  return ['1']; // Mocked worn items
                default:
                  return null;
              }
            }),
          },
        ],
      }),
      close: jest.fn(),
    }),
  }),
}));

// Mocking AuthContext
const mockAuthContext = {
  sessionToken: 'mockSessionToken',
};

// Mocking clothes_list to include an item with _id = 1
const mockClothesList = [
  { _id: '2', category: 'bottoms', name: 'jeans', image: require('../assets/clothing/jeans.png') },
  { _id: '9', category: 'tops', name: 'blue Skribble shirt', image: require('../assets/clothing/blueSkribbleShirt.png') },
  // Add more items as needed
];

// tests
describe('ClosetPage', () => {
  it('renders correctly and shows categories', async () => {
    const { getByText, getByTestId } = render(
      <AuthContext.Provider value={mockAuthContext}>
        <ClosetPage />
      </AuthContext.Provider>
    );

    // Check if the page renders correctly
    expect(getByText('Closet')).toBeTruthy();

    // Check if categories are rendered
    category_list.forEach(category => {
      expect(getByTestId(`categoryText-${category.category_id}`)).toBeTruthy();
    });
  });

  it('selects a category and shows corresponding items', async () => {
    const { getByTestId } = render(
      <AuthContext.Provider value={mockAuthContext}>
        <ClosetPage />
      </AuthContext.Provider>
    );

    // Select a category
    const bottomsCategory = getByTestId('category-bottoms');
    fireEvent.press(bottomsCategory);

    // Check if items from the selected category and owned items are rendered
    const ownedItems = ['2', '9']; // Mocked owned items
    const filteredItems = clothes_list.filter(item => item.category === 'bottoms' && ownedItems.includes(item._id));
    
    await waitFor(() => {
      expect(filteredItems.length).toBeGreaterThan(0);
    });
  });

  it('handles item selection and toggles wear status', async () => {
    const { getByTestId } = render(
      <AuthContext.Provider value={mockAuthContext}>
        <ClosetPage />
      </AuthContext.Provider>
    );

    // Select an item
    const item = clothes_list[0]; // Example item
    const itemButton = getByTestId(`item-${item._id}`);
    fireEvent.press(itemButton);

    // Simulate the item being worn and check if the worn item appears on the avatar
    expect(getByTestId(`wornItem-${item._id}`)).toBeTruthy();

    // Simulate item being removed (toggle the worn state)
    fireEvent.press(itemButton);
    expect(() => getByTestId(`wornItem-${item._id}`)).toThrow();
  });

  it('fetches and displays user data correctly from the database', async () => {
    const { getByTestId } = render(
      <AuthContext.Provider value={mockAuthContext}>
        <ClosetPage />
      </AuthContext.Provider>
    );

    // Wait for user data to load
    await waitFor(() => getByTestId('avatarImage'));
  });
});