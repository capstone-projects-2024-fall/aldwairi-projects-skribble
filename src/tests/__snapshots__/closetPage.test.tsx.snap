// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClosetPage renders correctly 1`] = `
<RCTScrollView
  style={
    [
      {
        "flex": 1,
        "padding": 18,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#ebebeb",
            "borderRadius": 54,
            "borderWidth": 2.7,
            "marginLeft": 45,
            "marginRight": 45,
            "opacity": 1,
            "padding": 13.5,
          }
        }
      >
        <Text
          style={
            {
              "color": "#0000000",
              "fontSize": 19.8,
              "fontWeight": "bold",
            }
          }
        >
          Back
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            {
              "height": 101.25,
              "maxHeight": 54,
              "width": 202.5,
            }
          }
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#ebebeb",
            "borderRadius": 54,
            "borderWidth": 2.7,
            "marginLeft": 45,
            "marginRight": 45,
            "opacity": 1,
            "padding": 13.5,
          }
        }
      >
        <Text
          style={
            {
              "color": "#0000000",
              "fontSize": 19.8,
              "fontWeight": "bold",
            }
          }
        >
          Store
        </Text>
      </View>
    </View>
    <Text
      style={
        {
          "color": "#000",
          "fontSize": 21.6,
          "fontWeight": "bold",
          "marginBottom": 9,
          "marginTop": -13.5,
          "marginVertical": 4.5,
          "textAlign": "center",
        }
      }
    >
      Closet
    </Text>
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "height": 533.6,
            "justifyContent": "center",
            "position": "relative",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            {
              "height": 207,
              "marginTop": 135,
              "width": 207,
              "zIndex": 0,
            }
          }
        />
        <Image
          resizeMode="contain"
          style={
            {
              "height": 225,
              "position": "absolute",
              "top": 0,
              "width": 225,
              "zIndex": 1,
            }
          }
        />
      </View>
      <View
        style={
          {
            "backgroundColor": "#5C2E00",
            "borderRadius": 10,
            "borderWidth": 3.6,
            "flex": 1,
          }
        }
      >
        <View
          style={
            {
              "borderBottomWidth": 3.6,
              "borderRadius": 10,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-around",
              "marginBottom": 9,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 63,
                  "marginBottom": 4.5,
                  "width": 63,
                }
              }
            />
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 10.8,
                  "fontWeight": "bold",
                  "marginBottom": 9,
                  "textAlign": "center",
                }
              }
            >
              bottoms
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 63,
                  "marginBottom": 4.5,
                  "width": 63,
                }
              }
            />
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 10.8,
                  "fontWeight": "bold",
                  "marginBottom": 9,
                  "textAlign": "center",
                }
              }
            >
              tops
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 63,
                  "marginBottom": 4.5,
                  "width": 63,
                }
              }
            />
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 10.8,
                  "fontWeight": "bold",
                  "marginBottom": 9,
                  "textAlign": "center",
                }
              }
            >
              full outfits
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 63,
                  "marginBottom": 4.5,
                  "width": 63,
                }
              }
            />
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 10.8,
                  "fontWeight": "bold",
                  "marginBottom": 9,
                  "textAlign": "center",
                }
              }
            >
              shoes
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 63,
                  "marginBottom": 4.5,
                  "width": 63,
                }
              }
            />
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 10.8,
                  "fontWeight": "bold",
                  "marginBottom": 9,
                  "textAlign": "center",
                }
              }
            >
              accessories
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-around",
              "paddingLeft": 9,
              "paddingRight": 9,
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
